### PR TITLE
Add Swift Package Manager and a few tests

### PR DESCRIPTION
**Pull request by distressedrook**
Tested on Xcode. All tests are passing.

### CODE REVIEW

**Comment by TiagoMaiaL** [comment]
@avismarahl

I believe we should keep this PR as small as possible, so it can easily get reviewed by the maintainers. As an initial strategy, we can only include the manifest file, configured with the current folder structure. That being said, I believe we could do the following:
- Remove the test target and its configuration in this first PR (including the test folders)
- Keep the current folder structure unchanged (configure the `target path` in the manifest)
- Fix any compilation errors (the test code shouldn't be at the `top level`, so we could just comment them)

I also noticed that the `xcuserdatad` folder is in the commits, we need to first ignore it. I'll submit a PR with this additions to the `.gitignore`.

I also have some considerations regarding the manifest file, I'll post them as comments in the source code.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// swift-tools-version:5.3
+// The swift-tools-version declares the minimum version of Swift required to build this package.
```
Comment (TiagoMaiaL) [inline Package.swift:2]: We can remove the comments (except the first one).

**Comment by carloshmartins** [comment]
@TiagoMaiaL We could close this PR due to inactivity, right? You reviewed the code, but nothing had been done from PR's owner

**Comment by TiagoMaiaL** [comment]
@carloshmartins Yes, I think we can close this PR. If you think it's worth it to add SPM support, I can still work on this. Let me know in case you do.

**Comment by anantcodes** [comment]
@carloshmartins I think we can close this for now.

**Pull request closed without merging**